lib/message.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'pg'
+
+# set up class which all posts/tweets have
+class Message
+  attr_reader :outputs
+
+  def self.all
+    outputs = []
+    connection = if ENV['ENVIRONMENT'] == 'test'
+                   PG.connect(dbname: 'messages_app_test')
+                 else
+                   PG.connect(dbname: 'messages_app')
+                 end
+
+    result = connection.exec('SELECT * FROM messages ORDER BY user_id DESC;')
+    result.each do |message|
```

Comment:
This is fine but it would be neater if you used `#map`

```ruby
result.map do |message|
  { name: message['name], message: message['message'], time: message['time'] }
end
```

Map always returns a new array, so this one line would replace 8.

Reply:
OK three lines! :)